extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -139,5 +313,236 @@ public void registerSwaggerUiServletExtension(SwaggerUiRecorder recorder,
+
+        /**
+         * If set, limits the number of tagged operations displayed to at most this many. The default is to show all operations.
+         */
+        @ConfigItem(name = "maxDisplayedTags")
+        Optional<Integer> maxDisplayedTags;
+
+        /**
+         * Apply a sort to the operation list of each API.
+         * It can be 'alpha' (sort by paths alphanumerically), 'method' (sort by HTTP method) or a function (see
+         * Array.prototype.sort() to know how sort function works).
+         * Default is the order returned by the server unchanged.
+         */
+        @ConfigItem(name = "operationsSorter")
+        Optional<String> operationsSorter;
+
+        /**
+         * Controls the display of vendor extension (x-) fields and values for Operations, Parameters, and Schema.
+         */
+        @ConfigItem(name = "showExtensions")
+        Optional<Boolean> showExtensions;
+
+        /**
+         * Controls the display of extensions (pattern, maxLength, minLength, maximum, minimum) fields and values for
+         * Parameters.
+         */
+        @ConfigItem(name = "showCommonExtensions")
+        Optional<Boolean> showCommonExtensions;
+
+        /**
+         * Apply a sort to the tag list of each API.
+         * It can be 'alpha' (sort by paths alphanumerically) or a function (see Array.prototype.sort() to learn how to write a
+         * sort function).
+         * Two tag name strings are passed to the sorter for each pass. Default is the order determined by Swagger UI.
+         */
+        @ConfigItem(name = "tagsSorter")
+        Optional<String> tagsSorter;
+
+        /**
+         * Provides a mechanism to be notified when Swagger UI has finished rendering a newly provided definition.
+         */
+        @ConfigItem(name = "onComplete")
+        Optional<String> onComplete;
+
+        /**
+         * Set to false to deactivate syntax highlighting of payloads and cURL command, can be otherwise an object with the
+         * activate and theme properties.
+         */
+        @ConfigItem(name = "syntaxHighlight")
+        Optional<String> syntaxHighlight;
+
+        /**
+         * OAuth redirect URL.
+         */
+        @ConfigItem(name = "oauth2RedirectUrl")
+        Optional<String> oauth2RedirectUrl;
+
+        /**
+         * MUST be a function. Function to intercept remote definition, "Try it out", and OAuth 2.0 requests.
+         * Accepts one argument requestInterceptor(request) and must return the modified request, or a Promise that resolves to
+         * the modified request.
+         */
+        @ConfigItem(name = "requestInterceptor")
+        Optional<String> requestInterceptor;
+
+        /**
+         * If set, MUST be an array of command line options available to the curl command.
+         * This can be set on the mutated request in the requestInterceptor function.
+         */
+        @ConfigItem(name = "requestCurlOptions")
+        Optional<String[]> requestCurlOptions;
```

Comment:
Make it a `List<String>` ?

Reply:
I think both String[] and List<String> will be the same in application.properties ? I still want it Optional, because the default is actually nothing (i.e do not add this in the html)

Reply:
> I think both String[] and List will be the same in application.properties ? 

Yes, it should. Although I think because a bug you need to do `Optional<List<String>>` instead

> I still want it Optional, because the default is actually nothing (i.e do not add this in the html)

That's fine, see above. You can also skip it if the list is empty though